File: Python3/next_greater_element_I.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result = []
        
        for val1 in nums1:
            index = nums2.index(val1)
            for i in range(index, len(nums2)):
                if nums2[i] > val1:
                    result.append(nums2[i])
                    break
            else: 
                result.append(-1)
        
        return result

File: C/next_greater_element_I.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int find(int* nums, int size, int x){
    int res;
    
    for(int i = 0; i < size; i++){
        if(nums[i] == x){
            res = i;
            break;
        }
    } 
        
    return res;
}
    
int* nextGreaterElement(int* nums1, int nums1Size, int* nums2, int nums2Size, int* returnSize){
    *returnSize = nums1Size;
    int* result = malloc(*returnSize * sizeof(int));
        
    for(int i = 0; i < nums1Size; i++){
        int tmp = find(nums2, nums2Size, nums1[i]);
        bool no = true;
        for(int j = tmp; j < nums2Size; j++){
            if(nums2[j] > nums1[i]){
                result[i] = nums2[j];
                no = false;
                break;
            }
        }
        if(no) result[i] = -1;
    }
        
    return result;
}
